core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1903,11 +1900,13 @@ protected Type visitExists(ExistsPredicate node, StackableAstVisitorContext<Cont
         @Override
         protected Type visitQuantifiedComparisonExpression(QuantifiedComparisonExpression node, StackableAstVisitorContext<Context> context)
         {
+            quantifiedComparisons.add(NodeRef.of(node));
+
             Expression value = node.getValue();
             process(value, context);
 
             Expression subquery = node.getSubquery();
-            process(subquery, context);
+            setExpressionType(subquery, analyzeSubquery((SubqueryExpression) subquery, context));
```

Comment:
`analyzeSubquery` is always accompanied by `setExpressionType`. Could `analyzeSubquery` do 
```
return(setExpressionType(...))
```
instead?

Reply:
Yes, that's leftover from a previous experiment.